person.rb

```diff
@@ -23,3 +30,33 @@ def of_age?
     @age > 18
   end
 end
+
+class Decorator < Person
+  def initialize(person)
+    super
+    @person = person
+  end
+
+  def correct_name
+    @person.name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @person.name.upcase
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @person.name[0...10]
```

Comment:
- [ ] I would implore you to do a proper check before you trim the name. What you want to achieve is for this to get trimmed if it is more than 10 characters. You need to check two things:
- Are the letters more than 10? To achieve this, remember array is index based, so it starts from 0 so to get 1 to 10, we will need to reach 9.
- We need a condition that will be like `@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`. Let me know if you need further clarification.